bootstrap.js

```diff
@@ -39,25 +39,34 @@ module.exports = function(grunt) {
     tasksDefault.push('validate');
   }
 
-  // If build/html exists, but is empty, skip the newer check.
-  // This facilitates situations where the build/html is generated as a mounted
-  // directory point with a newer timestamp than the Drush Makefiles.
-  //
-  // We do not use the grunt-newer .cache with drushmake so skipping newer for
-  // any one run does not impact later behavior.
-  if (grunt.file.exists(grunt.config.get('config.buildPaths.html') + '/index.php')) {
-    tasksDefault.push('newer:drushmake:default');
-  }
-  else {
-    tasksDefault.push('drushmake:default');
+  // Process .make files if configured.
+  if (grunt.config.get('config.srcPaths.make')) {
+    // If build/html exists, but is empty, skip the newer check.
+    // This facilitates situations where the build/html is generated as a mounted
+    // directory point with a newer timestamp than the Drush Makefiles.
+    //
+    // We do not use the grunt-newer .cache with drushmake so skipping newer for
+    // any one run does not impact later behavior.
+    if (grunt.file.exists(grunt.config.get('config.buildPaths.html') + '/index.php')) {
+      tasksDefault.push('newer:drushmake:default');
+    }
+    else {
+      tasksDefault.push('drushmake:default');
+    }
   }
 
   // Wire up the generated docroot to our custom code.
   tasksDefault.push('scaffold');
 
-  if (grunt.config.get(['composer', 'install'])) {
+  if (grunt.file.exists('./composer.lock') && grunt.config.get(['composer', 'install'])) {
+    // Run `composer install` if there is already a lock file. Updates should be explicit once this file exists.
     tasksDefault.unshift('composer:install');
```

Comment:
We need to manually run `composer drupal-scaffold` here too, *or* go figure out why Drupal Scaffold removed it.